### PR TITLE
First pass at my calendar feature

### PR DESCRIPTION
**Pull request by sirowland**

### CODE REVIEW

File: client/src/components/App.jsx [hunk 1]
```diff
@@ -1,10 +1,81 @@
...
+    this.getListing();
+    this.getCalendar();
+  }
+
+  getListing() {
+    axios.get(`http://localhost:3001/api/listings/${this.state.listingId}`)
+      .then((res) => {
+        const { data } = res;
+        this.setState({
+          listingId: data.id,
```
Comment (ruslanfarutdinov) [inline client/src/components/App.jsx]: Since `this.state.listingId` is hardcoded to `1001`, the response will always be for a listing with id 1001 -  is that correct? If so, how will you handle different `listingId`'s coming from the client?

File: client/src/components/Calendar.jsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Date from './Date.jsx';
+
+const Calendar = (props) => {
```
Comment (ruslanfarutdinov) [inline client/src/components/Calendar.jsx]: Perhaps could use destructuring here? `({ daysUnav, etc })`

File: client/src/index.jsx [hunk 3]
```diff
@@ -1,5 +1,8 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
-import App from './components/App';
+import App from './components/App.jsx';
 
-ReactDOM.render(<App />, document.getElementById('app'));
+ReactDOM.render(
+  <App />,
+  document.getElementById('app'),
+);
```
Comment (ruslanfarutdinov) [inline client/src/index.jsx]: I'm curious, why the change to multiple lines? Readability?

File: database/index.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    EXTRACT(DAY from date_booked) as day
+    FROM unavailable_dates 
+    WHERE listing_id = ${listingId}
+    AND EXTRACT(YEAR_MONTH from date_booked) = ${yearMonth}
+    ORDER BY date_booked ASC
+  `;
+
+  return connection
+    .then(conn => conn.query(sql))
+    .catch(err => err);
```
Comment (ruslanfarutdinov) [inline database/index.js]: Good choice on using `promise-mysql` - makes the code much cleaner!

File: server/app.js [hunk 5]
```diff
@@ -1,8 +1,58 @@
...
 const path = require('path');
+const bodyParser = require('body-parser');
+const db = require('../database');
+const dateHelpers = require('./dateHelpers');
 
 const app = express();
 
-app.use(express.static(path.join(__dirname, '../client/dist')));
+app.use('/', express.static(path.join(__dirname, '../client/dist')));
+app.use(bodyParser.json());
```
Comment (ruslanfarutdinov) [inline server/app.js]: Why use `bodyParser`?  Is there content coming in, in the body of the request?

File: server/app.js [hunk 6]
```diff
@@ -1,8 +1,58 @@
...
+
+app.get('/listings/:listingId', (req, res) => {
+  res.redirect('/');
+});
```
Comment (ruslanfarutdinov) [inline server/app.js]: Instead of redirecting the request to `/`, couldn't you just have the `/listings/:listingId` path on line 9 where you're serving static files? Lines 12-14 wouldn't be neccessary in that case.

File: server/app.js [hunk 7]
```diff
@@ -1,8 +1,58 @@
...
+
+app.get('/api/listings/:listingId', (req, res) => {
+  db.getListingInfo(req.params.listingId)
+    .then((rows) => {
+      res.send(rows[0]);
+    })
+    .catch((err) => {
+      res.statusCode(500).send(err);
```
Comment (ruslanfarutdinov) [inline server/app.js]: Nice error handling! Actually sending a status code and the error.

File: client/src/components/App.jsx (hunk 1 above)
Comment (sirowland) [reply client/src/components/App.jsx]: I am imagining that listingId will be passed in from the proxy server. This is just so I can test it.

File: client/src/components/Calendar.jsx (hunk 2 above)
Comment (sirowland) [reply client/src/components/Calendar.jsx]: oh nice didn't know I could do that!

File: client/src/index.jsx (hunk 3 above)
Comment (sirowland) [reply client/src/index.jsx]: no reason I will go back and change it in later versions

File: server/app.js (hunk 5 above)
Comment (sirowland) [reply server/app.js]: true, not sure I'll ever need it, thanks!

File: server/app.js (hunk 6 above)
Comment (sirowland) [reply server/app.js]: But then when they went to / wouldn't it error out? Not sure will test it out

File: client/src/components/App.jsx (hunk 1 above)
Comment (ruslanfarutdinov) [reply client/src/components/App.jsx]: Doing it for testing purpose makes sense. I am not 100% clear on how exactly the proxy server will pass down the `listingId` and your `this.state.listingId` will receive it, but I think that's something that we all will solve once we get to that point.

File: server/app.js (hunk 6 above)
Comment (ruslanfarutdinov) [reply server/app.js]: It would error out, but I assumed we shouldn't be taking care of the `/` endpoint, since we're building a product page and a product page has to have a product id; otherwise, if the client just enters `/`, there is no way of knowing which product s/he wanted to see.

Comment (sirowland) [reply server/app.js]: cool I'll take it out then
